professor.py

```diff
@@ -28,16 +26,35 @@ def main():
     print("#"*80)
     print('\t\t\tWelcome to classroom voter')
     print("#"*80)
+    
+    ip, port = prompt_for_ip()
+    clientSocket = socket.socket()
+    try:
+        clientSocket.connect((ip, port))
+        print('Successful Connection')
+    except socket.error as e:
+        print('Failed Connection: ' + str(e))
 
     poll = prompt_for_poll()
-    ip, port = prompt_for_ip()
     msg = {
         "endpoint": "Announce_poll",
         "Arguments": {
-            "poll": poll.toBytes()
+            "poll": poll.toDict()
```

Comment:
Converting to bytes in the json and then encoding the json before we send a message was problematic. To fix this I just made it a dict and then encode all of msg at the same time.